modules/microshift-install-rpms-olm.adoc

```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// microshift/microshift-install-rpm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-with-olm-from-rpm-package_{context}"]
+== Installing the Operator Lifecycle Manager (OLM) from an RPM package
+
+When you install {microshift-short}, the Operator Lifecycle Manager (OLM) package is not installed by default. You can install the OLM on your {microshift-short} instance using a RPM package. 
+
+.Prerequisites
+
+* You have created a {microshift-short} instance. 
+
+.Procedure 
+
+. Install the OLM package by running the following command: 
++
+[source,terminal]
+----
+$ sudo dnf install microshift-olm
```

Comment:
@kelbrown20 It appears the build of MicroShift I am using already has microshift-olm installed.

redhat@microshift-dev ~]$ rpm -qa | grep microshift-olm
microshift-olm-4.15.0_0.nightly_2024_01_25_051548_20240130175736_7c3d2e39e-1.el9.x86_64


 Can you check with development to verify this step is actually needed  ?

Reply:
@kevinrizza @pmtk Hello! Do you guys have any info on this, it seems like OLM is installed by default. Maybe this is for certain instances? Or should I say something instead like, 
`This is how you install OLM on MicroShift` and like `You can check if it is already installed with <command> command`

Reply:
@dhensel-rh what host is that? is it ostree? in jenkins? what do you have in kickstart?

Today I provisioned microshift using clusterbot and it didn't have microshift-olm installed. [Command](https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/microshift/clusterbot/prepare-host/openshift-microshift-clusterbot-prepare-host-commands.sh#L119C2-L119C50) it's using is more or less `dnf install microshift-${version}`.

Reply:
```
+ 14:13:54.248942564 /tmp/dnf_retry.SL76vvE4.sh:26 	sudo dnf install -y microshift-4.15.0~rc.4-202401262049.p0.gd05be5f.assembly.rc.4.el9
Updating Subscription Management repositories.
MicroShift mirror repository                    4.0 kB/s | 3.6 kB     00:00    
Dependencies resolved.
==============================================================================================================================================
 Package                            Arch    Version                                                Repository                             Size
==============================================================================================================================================
Installing:
 microshift                         x86_64  4.15.0~rc.4-202401262049.p0.gd05be5f.assembly.rc.4.el9 microshift-mirror                      50 M
Upgrading:
 NetworkManager                     x86_64  1:1.44.0-4.el9_3                                       rhel-9-baseos-rhui-rpms               2.3 M
 NetworkManager-cloud-setup         x86_64  1:1.44.0-4.el9_3                                       rhel-9-appstream-rhui-rpms             77 k
 NetworkManager-libnm               x86_64  1:1.44.0-4.el9_3                                       rhel-9-baseos-rhui-rpms               1.8 M
 NetworkManager-team                x86_64  1:1.44.0-4.el9_3                                       rhel-9-baseos-rhui-rpms                43 k
 NetworkManager-tui                 x86_64  1:1.44.0-4.el9_3                                       rhel-9-baseos-rhui-rpms               249 k
Installing dependencies:
 NetworkManager-ovs                 x86_64  1:1.44.0-4.el9_3                                       rhel-9-appstream-rhui-rpms             64 k
 bubblewrap                         x86_64  0.4.1-6.el9                                            rhel-9-for-x86_64-baseos-rpms          52 k
 conmon                             x86_64  3:2.1.7-3.1.rhaos4.14.el9                              rhocp-4.14-for-rhel-9-x86_64-rpms      50 k
 conntrack-tools                    x86_64  1.4.7-2.el9                                            rhel-9-for-x86_64-appstream-rpms      239 k
 container-selinux                  noarch  3:2.224.0-1.rhaos4.14.el9                              rhocp-4.14-for-rhel-9-x86_64-rpms      51 k
 containernetworking-plugins        x86_64  1:1.3.0-6.el9_3                                        rhel-9-appstream-rhui-rpms            9.3 M
 containers-common                  x86_64  2:1-55.el9                                             rhel-9-appstream-rhui-rpms            138 k
 cri-o                              x86_64  1.27.3-2.rhaos4.14.git03502b6.el9                      rhocp-4.14-for-rhel-9-x86_64-rpms      33 M
 cri-tools                          x86_64  1.27.0-2.1.el9                                         rhocp-4.14-for-rhel-9-x86_64-rpms     9.2 M
 criu                               x86_64  3.18-1.el9                                             rhel-9-appstream-rhui-rpms            555 k
 crun                               x86_64  1.12-1.rhaos4.14.el9                                   rhocp-4.14-for-rhel-9-x86_64-rpms     210 k
 fuse                               x86_64  2.9.9-15.el9                                           rhel-9-for-x86_64-baseos-rpms          84 k
 fuse-common                        x86_64  3.10.2-6.el9                                           rhel-9-baseos-rhui-rpms               8.5 k
 fuse3                              x86_64  3.10.2-6.el9                                           rhel-9-appstream-rhui-rpms             57 k
 fuse3-libs                         x86_64  3.10.2-6.el9                                           rhel-9-appstream-rhui-rpms             93 k
 greenboot                          x86_64  0.15.4-1.el9                                           rhel-9-for-x86_64-appstream-rpms       41 k
 jq                                 x86_64  1.6-15.el9                                             rhel-9-appstream-rhui-rpms            190 k
 libnet                             x86_64  1.2-6.el9                                              rhel-9-for-x86_64-appstream-rpms       61 k
 libnetfilter_cthelper              x86_64  1.0.0-22.el9                                           rhel-9-for-x86_64-appstream-rpms       26 k
 libnetfilter_cttimeout             x86_64  1.0.0-19.el9                                           rhel-9-for-x86_64-appstream-rpms       25 k
 libnetfilter_queue                 x86_64  1.0.5-1.el9                                            rhel-9-for-x86_64-appstream-rpms       31 k
 libslirp                           x86_64  4.4.0-7.el9                                            rhel-9-for-x86_64-appstream-rpms       72 k
 microshift-greenboot               noarch  4.15.0~rc.4-202401262049.p0.gd05be5f.assembly.rc.4.el9 microshift-mirror                      20 k
 microshift-networking              x86_64  4.15.0~rc.4-202401262049.p0.gd05be5f.assembly.rc.4.el9 microshift-mirror                      19 k
 microshift-selinux                 noarch  4.15.0~rc.4-202401262049.p0.gd05be5f.assembly.rc.4.el9 microshift-mirror                      25 k
 oniguruma                          x86_64  6.9.6-1.el9.5                                          rhel-9-for-x86_64-appstream-rpms      221 k
 openvswitch-selinux-extra-policy   noarch  1.0-34.el9fdp                                          fast-datapath-for-rhel-9-x86_64-rpms   14 k
 openvswitch3.1                     x86_64  3.1.0-73.el9fdp                                        fast-datapath-for-rhel-9-x86_64-rpms  6.9 M
 ostree                             x86_64  2023.6-1.el9                                           rhel-9-appstream-rhui-rpms            294 k
 ostree-libs                        x86_64  2023.6-1.el9                                           rhel-9-appstream-rhui-rpms            472 k
 protobuf-c                         x86_64  1.3.3-13.el9                                           rhel-9-baseos-rhui-rpms                37 k
 python3-pexpect                    noarch  4.8.0-7.el9                                            rhel-9-for-x86_64-baseos-rpms         155 k
 python3-ptyprocess                 noarch  0.6.0-12.el9                                           rhel-9-for-x86_64-baseos-rpms          35 k
 rpm-ostree                         x86_64  2023.7-1.el9                                           rhel-9-appstream-rhui-rpms            4.6 M
 rpm-ostree-libs                    x86_64  2023.7-1.el9                                           rhel-9-appstream-rhui-rpms             24 k
 runc                               x86_64  4:1.1.10-1.rhaos4.14.el9                               rhocp-4.14-for-rhel-9-x86_64-rpms     3.1 M
 sos                                noarch  4.6.0-5.el9                                            rhel-9-for-x86_64-baseos-rpms         1.1 M
 unbound-libs                       x86_64  1.16.2-3.el9                                           rhel-9-for-x86_64-appstream-rpms      553 k
 yajl                               x86_64  2.1.0-22.el9                                           rhel-9-appstream-rhui-rpms             42 k
Installing weak dependencies:
 criu-libs                          x86_64  3.18-1.el9                                             rhel-9-appstream-rhui-rpms             32 k
 fuse-overlayfs                     x86_64  1.12-1.el9                                             rhel-9-appstream-rhui-rpms             69 k
 skopeo                             x86_64  2:1.13.3-3.el9_3                                       rhel-9-appstream-rhui-rpms            7.9 M
 slirp4netns                        x86_64  1.2.1-1.el9                                            rhel-9-appstream-rhui-rpms             49 k
```

Reply:
I installed using clusterbot 
`workflow-launch openshift-microshift-clusterbot 4.15  "MICROSHIFT_GIT=release-4.15"`

[ec2-user@i-0d464fbc965691373 ~]$ rpm -qa | grep olm
microshift-olm-4.15.0_0.nightly_2024_01_30_205726_20240131171500_962c4da72-1.el9.x86_64

[ec2-user@i-0d464fbc965691373 ~]$ rpm-ostree status
error: This system was not booted via libostree.
Currently, most rpm-ostree commands only work on ostree-based host systems.

Reply:
This is because you've built from source (MICROSHIFT_GIT) - in such case `configure-vm.sh` [installs all RPM that were built](https://github.com/openshift/microshift/blob/main/scripts/devenv-builder/configure-vm.sh#L212).
If you'd install them from remote repository, using just `dnf install microshift`, the microshift-olm package will not be installed.
You can check by using `workflow-launch openshift-microshift-clusterbot 4.15` which will get RPMs from the mirror (rc4 now)

Reply:
Thanks for the clarification @pmtk

Reply:
@dhensel-rh Thank you for talking a look Douglas! Does everything else look good otherwise?